### PR TITLE
Detect wider variety of usernames for SSH-based remotes

### LINKED ISSUE #21507: Failure to open permalink to line for organization-specific SSH username remotes
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

My work's organization using GitHub has a specific username for cloning repositories other than `git`. I can't find any clear cut documentation on what this is used for, but see attached screenshot for an example:

clone instructions

<br>
<br>

When executing the "editor: open permalink to line" command, it fails with:

Failed to open permalink: failed to parse Git remote URL

### Environment

Zed: v0.163.3 (Zed)
OS: macOS 15.1.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```

```
</details>

### PR DESCRIPTION
**Pull request by nickbreaton**
Closes #21507

Release Notes:

- Fixed detection of git remotes when using SSH and username is not "git".

### CODE REVIEW

**Comment by nickbreaton** [comment]
@cla-bot check

**Comment by nickbreaton** [comment]
This is my first experience contributing to this project, and my first Rust PR. Happy to accommodate any and all feedback!

File: crates/git/Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,7 @@ text.workspace = true
 time.workspace = true
 url.workspace = true
 util.workspace = true
+regex.workspace = true
```
Comment (osiewicz) [inline crates/git/Cargo.toml:33]: Let's put this dep on line 25 to keep the list of dependencies sorted.

**Review by osiewicz — APPROVED** [review]
LGTM

File: crates/git/Cargo.toml (hunk 1 above)
Comment (nickbreaton) [reply crates/git/Cargo.toml]: done 🙂

**Pull request merged**